src/lib.rs

```diff
@@ -90,282 +85,142 @@ macro_rules! implement {
 implement!(LowerHex, "{:02x}");
 implement!(UpperHex, "{:02X}");
 
-/// A context.
-#[derive(Clone)]
-pub struct Context {
-    buffer: [u8; 64],
-    count: [u32; 2],
-    state: [u32; 4],
-}
+// Constants
+const PADDING: [u8; 64] = {
+    let mut padding = [0; 64];
```

Comment:
To not have padding in padding, just call it `data`.

Reply:
Done